components/brave_origin/brave_origin_service.cc

```diff
@@ -70,61 +88,50 @@ bool BraveOriginService::SetBrowserPolicyValue(std::string_view pref_name,
   }
 
   // Set the policy value in BraveOriginPolicyManager
-  manager->SetBrowserPolicyValue(pref_name, value);
-
-  // Also set the corresponding pref value in local_state
-  // If not user-settable and value equals default, clear the pref instead
-  // We don't want a user to have a value set in local state if the value is not
-  // user settable as they would be stuck with that value if they stopped being
-  // a Brave Origin user
-  if (!pref_info->user_settable && value == pref_info->default_value) {
-    local_state_->ClearPref(pref_info->pref_name);
-  } else {
-    local_state_->SetBoolean(pref_info->pref_name, value);
+  PrefService* target_prefs = nullptr;
+  if (manager->IsBrowserPolicy(pref_info->pref_name)) {
+    manager->SetPolicyValue(pref_name, value);
+    target_prefs = local_state_;
+  } else if (manager->IsProfilePolicy(pref_info->pref_name)) {
+    manager->SetPolicyValue(pref_name, value, profile_id_);
+    target_prefs = profile_prefs_;
   }
 
-  return true;
-}
-
-bool BraveOriginService::SetProfilePolicyValue(std::string_view pref_name,
-                                               bool value) {
-  if (!IsBraveOriginEnabled()) {
-    return false;
+  // Also set the corresponding pref value
+  if (target_prefs) {
+    if (!pref_info->user_settable && value == pref_info->default_value) {
+      target_prefs->ClearPref(pref_info->pref_name);
+    } else {
+      target_prefs->SetBoolean(pref_info->pref_name, value);
+    }
   }
 
-  // Get policy info to access pref_key and user_settable
-  auto* manager = BraveOriginPolicyManager::GetInstance();
-  const BraveOriginPolicyInfo* pref_info = manager->GetPrefInfo(pref_name);
-  if (!pref_info) {
-    return false;
+  // Reload browser policies to ensure the new policy value takes effect
+  if (manager->IsBrowserPolicy(pref_info->pref_name) &&
+      browser_policy_service_) {
+    browser_policy_service_->RefreshPolicies(
```

Comment:
Would it make sense to notify the providers via the observer interface so they can just call `UpdatePolicy` instead of refreshing all policies? IIRC, this might be more efficient.